### PR TITLE
MNT: Use select for entry_points

### LINKED ISSUE #11524: TST: DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
Example log: https://github.com/astropy/astropy/runs/2321021984?check_suite_focus=true

Snippet (one of many): 

```
___________ ERROR collecting astropy/modeling/tests/data/__init__.py ___________
tests/lib/python3.7/site-packages/pytest_doctestplus/plugin.py:206: in collect
    module = self.fspath.pyimport()
tests/lib/python3.7/site-packages/py/_path/local.py:704: in pyimport
    __import__(modname)
astropy/modeling/__init__.py:10: in <module>
    from . import fitting
astropy/modeling/fitting.py:1764: in <module>
    populate_entry_points(entry_points().get('astropy.modeling', []))
tests/lib/python3.7/site-packages/importlib_metadata/__init__.py:312: in get
    flake8_bypass(self._warn)()
E   DeprecationWarning: SelectableGroups dict interface is deprecated. Use select.
```

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to replace deprecated `.get` with `.select`.

Fix #11524 

Follow-up of #11091

### CODE REVIEW

File: astropy/io/misc/asdf/tests/__init__.py [hunk 1]
```diff
@@ -11,10 +11,15 @@
 
 import pytest
 
-ep = [entry.name for entry in entry_points().get('asdf_extensions', [])]
+# TODO: Exclusively use select when Python minversion is 3.10
+eps = entry_points()
+if hasattr(eps, 'select'):
+    ep = [entry.name for entry in eps.select('asdf_extensions', [])]
```
Comment (saimn) [inline astropy/io/misc/asdf/tests/__init__.py:17]:
```suggestion
    ep = [entry.name for entry in eps.select(group='asdf_extensions')]
```

File: astropy/modeling/fitting.py [hunk 2]
```diff
@@ -1761,4 +1761,13 @@ def populate_entry_points(entry_points):
                         'astropy.modeling.Fitter' .format(name)))
 
 
-populate_entry_points(entry_points().get('astropy.modeling', []))
+def _populate_ep():
+    # TODO: Exclusively use select when Python minversion is 3.10
+    ep = entry_points()
+    if hasattr(ep, 'select'):
+        populate_entry_points(ep.select('astropy.modeling', []))
```
Comment (saimn) [inline astropy/modeling/fitting.py:1768]:
```suggestion
        populate_entry_points(ep.select(group='astropy.modeling'))
```

**Review by saimn — COMMENTED** [review]
The API is different with the new version.

**Comment by pllim** [comment]
Thanks for the fix. You caught me not RTFM. 😅

**Comment by pllim** [comment]
@saimn , I don't think the failures are related. Do you want to approve?

**Review by saimn — APPROVED** [review]
Thanks @pllim !

**Comment by pllim** [comment]
Thank YOU, @saimn ! 😉

**Pull request merged**